### PR TITLE
dygma/raise: multiple fixes and improvements

### PR DESCRIPTION
**Pull request by justfalter**
This PR makes multiple fixes and improvements to the dygma raise implementation. Much of this centered around improving the accuracy and performance of LED light handling. 

- respect RGB_MATRIX_ENABLE = no
- Only flush LED buffer when changes have occurred. This significantly improves typing accuracy by reducing traffic related to LED state, increasing the bandwidth available for matrix scans.
- improve configurability of RGB matrix, reduce refresh rate
  - Reduce LED refresh rate from 60fps to 10fps, given the significant impact on typing accurate.
  - Allow enabling/disabling of reactive RGB matrix modes from rules.mk
- fix right-hand ansi LED key map
  - The ansi layout right hand LED key mapping was seemingly flipped and offset by 1 key. For example, when the LED for "7" was to light, the "= +" LED would light, instead. Reversing each row of the led mapping resolved the issue.
  - LEDs for the "\ |" and "enter" keys were swapped.
- fix color order to GRB
  - Colors were incorrectly mapped with red and green swapped. Actual color order is GRB, not RGB.
- Add color-correction for red channel, as it apparently receives more voltage than the other color channels.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by justfalter** [comment]
@ibash you might be interested in this

**Comment by ibash** [comment]
Awesome! Lgtm

Out of curiosity what are you using for a mcu?

**Review by drashna — APPROVED** [review]

File: keyboards/handwired/dygma/raise/leds.c [hunk 1]
```diff
@@ -14,6 +14,8 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#ifdef RGB_MATRIX_ENABLE
+
```
Comment (fauxpark) [inline keyboards/handwired/dygma/raise/leds.c:18]:
```suggestion
```
Use `post_rules.mk` to control whether this code is compiled.

File: keyboards/handwired/dygma/raise/leds.c [hunk 2]
```diff
@@ -99,3 +100,5 @@ static void flush(void) {
 }
 
 const rgb_matrix_driver_t rgb_matrix_driver = {.init = init, .flush = flush, .set_color = set_color, .set_color_all = set_color_all};
+
+#endif
```
Comment (fauxpark) [inline keyboards/handwired/dygma/raise/leds.c:104]:
```suggestion
```

File: keyboards/handwired/dygma/raise/leds.h [hunk 3]
```diff
@@ -15,11 +15,12 @@
  */
 
 #pragma once
+#ifdef RGB_MATRIX_ENABLE
```
Comment (fauxpark) [inline keyboards/handwired/dygma/raise/leds.h:18]:
```suggestion
```

File: keyboards/handwired/dygma/raise/leds.h [hunk 4]
```diff
@@ -15,11 +15,12 @@
...
 
 #include "quantum.h"
 #include "rgb_matrix.h"
 
 extern const uint8_t led_map[DRIVER_LED_TOTAL];
 
-void set_all_leds_to(uint8_t r, uint8_t g, uint8_t b);
-void set_led_to(int led, uint8_t r, uint8_t g, uint8_t b);
+
+#endif
```
Comment (fauxpark) [inline keyboards/handwired/dygma/raise/leds.h:26]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by justfalter** [comment]
@fauxpark I got the changes in to use `post_rules.mk` instead of `#ifdef``'s . 

I also fixed dygma/raise's q_led_config, which was failing to validate (#17105).

**Comment by justfalter** [comment]
@drashna is this something you can take care of merging? I've addressed @fauxpark's requested changes.

**Comment by ibash** [comment]
@justfalter thanks for all the work here! Hopefully this can get a second approval soon...

**Review by ibash — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**